dagshub/upload/wrapper.py

```diff
@@ -24,6 +26,66 @@ def get_default_branch(src_url, owner, reponame, auth):
     return res.json().get('default_branch')
 
 
+def create_repo(repo_name, description="", private=False, auto_init=False, gitignores="", license="", readme="", template="notebook-template"):
+    import logging
+    logging.basicConfig(level=logging.DEBUG)
+    import dagshub.auth
+    from dagshub.auth.token_auth import HTTPBearerAuth
+
+    username = config.username
+    password = config.password
+    if username is not None and password is not None:
+        return username, password
+    try:
+        token = dagshub.auth.get_token(code_input_timeout=0)
+    except dagshub.auth.OauthNonInteractiveShellException:
+        logger.debug("Failed to perform OAuth in a non interactive shell")
+    if token is not None:
+        auth = HTTPBearerAuth(token)
+
+    userRes = requests.get(urllib.parse.urljoin(config.host, USER_INFO_URL), auth=auth)
+    userJson = userRes.json()
+    if username is None:
+        username = userJson["username"]
+
+    repoRes = requests.get(urllib.parse.urljoin(config.host, REPO_INFO_URL.format(
+        owner=username,
+        reponame=repo_name
+    )))
+    if repoRes.status_code == HTTPStatus.OK:
+        return Repo(owner=username, name=repo_name, username=username, token=token)
```

Comment:
You don't need any of this:
1. Just let the API return an error if the repo already exists
2. The full repo information, including owner name, gets returned in the JSON result from the create repo API
3. The desired owner might be an organization, as I'll describe in a comment below

Reply:
If we want a `get_or_create` functionality, that can be done with an additional helper function external to this one

Reply:
I'll leave the check for the repo, because the create just fails with no clear message that the repo already exists